src/cli/deploy.ts

```diff
@@ -10,12 +10,30 @@ import fetch from "cross-fetch";
 
 import { requireToken } from "./auth";
 import { loadConfig } from "./config";
-import { slugify } from "./utils";
 import PlaypassClient from "./playpass-client";
+import { walkDir } from "./utils";
 
 // TODO(2022-02-22): Put this in a project config or infer
 const PUBLISH_DIR = path.join(process.cwd(), "dist");
 
+async function walkPublishDir(publishDir: string): Promise<string[]> {
+    let files;
+    try {
+        files = await walkDir(publishDir);
+    } catch (e) {
+        throw new Error(`Could not read directory ${publishDir}. Try running 'npm run build' again.`);
+    }
+
+    if (files.length <= 0) {
+        throw new Error(`No files found in ${publishDir}. Try running 'npm run build' again.`);
+    }
+
+    if (!files.find(it => it === `${publishDir}/index.html`)) {
+        throw new Error(`Index file not found at ${publishDir}. Try running 'npm run build' again.`);
+    }
+    return files;
+}
+
```

Comment:
We already walk the filesystem in `packageDir()`, could we count the files there instead of doing a second walk?

It would also be a more accurate number, because we ignore certain hidden files during zip packaging.